src/modules/SX126x/STM32WLx_Module.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+
+Copyright (c) 2022 STMicroelectronics
+
+This file is licensed under the MIT License: https://opensource.org/licenses/MIT
+*/
+
+#include "STM32WLx_Module.h"
+
+#if !defined(RADIOLIB_EXCLUDE_STM32WLX)
+
+#include <SubGhz.h>
+
+// This defines some dummy pin numbers (starting at NUM_DIGITAL_PINS to
+// guarantee these are not valid regular pin numbers) that can be passed
+// to the parent Module class, to be stored here and then passed back to
+// the overridden callbacks when these are used.
+enum {
+  RADIOLIB_STM32WLx_VIRTUAL_PIN_NSS = NUM_DIGITAL_PINS,
+  RADIOLIB_STM32WLx_VIRTUAL_PIN_BUSY,
+  RADIOLIB_STM32WLx_VIRTUAL_PIN_IRQ,
+  RADIOLIB_STM32WLx_VIRTUAL_PIN_RESET,
+};
+
+
+STM32WLx_Module::STM32WLx_Module():
+  Module(
+    RADIOLIB_STM32WLx_VIRTUAL_PIN_NSS,
+    RADIOLIB_STM32WLx_VIRTUAL_PIN_IRQ,
+    RADIOLIB_STM32WLx_VIRTUAL_PIN_RESET,
+    RADIOLIB_STM32WLx_VIRTUAL_PIN_BUSY,
+    SubGhz.SPI,
+    SubGhz.spi_settings
+  )
+{
+  setCb_pinMode(virtualPinMode);
+  setCb_digitalWrite(virtualDigitalWrite);
+  setCb_digitalRead(virtualDigitalRead);
+}
+
+void STM32WLx_Module::virtualPinMode(uint32_t dwPin, uint32_t dwMode) {
+  switch(dwPin) {
+    case RADIOLIB_STM32WLx_VIRTUAL_PIN_NSS:
+    case RADIOLIB_STM32WLx_VIRTUAL_PIN_BUSY:
+    case RADIOLIB_STM32WLx_VIRTUAL_PIN_IRQ:
+    case RADIOLIB_STM32WLx_VIRTUAL_PIN_RESET:
+      // Nothing to do
+      break;
+    default:
+      // TODO: Should we save the original callbacks in the constructor and then call those
+      // instead of the real Arduino functions directly?
```

Comment:
It's probably not strictly necessary, unless you want to jump through all the hoops of generator macros in Module header. I don't think there's much chance of this being used outside of Arduino.

Reply:
Resolved in 02bd4293b7418210931bd40c543c4556e7984fcb.